studio/src/editors/version-panel.js

```diff
@@ -0,0 +1,701 @@
+        }
+    }
+
+    togglePanel() {
+        this.showPanel = !this.showPanel;
+    }
+
+    closePanel() {
+        this.showPanel = false;
+        this.showMenuFor = '';
+    }
+
+    handleClickOutside(event) {
+        if (this.showMenuFor && !event.target.closest('.version-actions-menu')) {
+            this.showMenuFor = '';
+        }
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        document.addEventListener('click', this.handleClickOutside.bind(this));
+    }
+
+    disconnectedCallback() {
+        super.disconnectedCallback();
+        document.removeEventListener('click', this.handleClickOutside.bind(this));
+    }
+
+    get backArrowIcon() {
+        return html`
+            <svg class="back-arrow-icon" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2">
+                <path d="M19 12H5M12 19l-7-7 7-7" />
+            </svg>
+        `;
+    }
+
+    get versionList() {
+        if (this.loading) {
+            return html`<div class="loading-message">Loading versions...</div>`;
+        }
+
+        if (!this.versions || this.versions.length === 0) {
+            return html`<div class="no-versions">No versions available</div>`;
+        }
+
+        return html`
+            <div class="version-list">
+                ${this.versions.map((version, index) => {
+                    const isSelected = version.id === this.selectedVersion;
+                    const isCurrent = index === 0; // First version is current
+                    return html`
+                        <div
+                            class="version-item ${isSelected ? 'selected' : ''} ${isCurrent ? 'current' : ''}"
+                            @click="${() => this.handleVersionClick(version)}"
+                        >
+                            <div class="version-header">
+                                <div class="version-title-section">
+                                    <div class="version-status">
+                                        ${isCurrent
+                                            ? html`
+                                                  <div class="current-indicator">
+                                                      <div class="current-dot"></div>
+                                                      Latest version
+                                                  </div>
+                                              `
+                                            : nothing}
+                                    </div>
+                                    <div class="version-label">${version.title || `Version ${version.version}`}</div>
+                                </div>
+                                <div class="version-actions-menu">
+                                    <button class="menu-trigger" @click="${(e) => this.toggleMenu(e, version)}">
+                                        <svg width="16" height="16" viewBox="0 0 24 24" fill="currentColor">
```

Comment:
@seanchoi-dev I still see many inline svg icons, can you confirm if you were not able to find the equivalent of these icons in spectrum web components?